### PR TITLE
Pop OIDN to v1.2.1, Use shutil.which to locate `odinDenoise` binary.

### PR DESCRIPTION
**Pull request by bartoszek**
Use `which()` to locate `oidnDenoise` binary, allowing use of system wide `openimagedenoise` e.g. on Linux.
Priority to local ( `cwd\bin` ) path.
Tested on:
- [x] Linux: {blender:2.83.1, openimagedenoise:1.2.0, luxcorerender-git:2.4.beta1.r1.g35eca7c19}
- [ ] Windows
- [ ] Mac

### CODE REVIEW

**Comment by Theverat** [comment]
To be honest, I don't really see a reason why we would want to use the system wide OIDN binary. 
It is not under our control, so it might be a different version than required and might not work in such a case.

**Comment by bartoszek** [comment]
For release standpoint, it makes no sense at all, but as the packager, I have multiple packages depending on `openimagedenoise` and have to patch `LuxBlendCore` to be able to work with system-wide `openimagedenoise`.
If it would be more acceptable, I could switch around ordering - prepending `CWD/bin` to make local `denoise` goes first :thinking:

**Comment by bartoszek** [comment]
Ok, now local path (cwd/bin) is the first to search.
Also I've advanced OIDN to v1.2.1.

File: draw/viewport.py [hunk 1]
```diff
@@ -73,9 +74,10 @@ def __init__(self, engine, context, scene):
         self._normal_file_path = self._make_denoiser_filepath("normal")
         self._denoised_file_path = self._make_denoiser_filepath("denoised")
         current_dir = os.path.dirname(os.path.realpath(__file__))
-        self._denoiser_path = os.path.join(os.path.dirname(current_dir), "bin", "denoise")
-        if platform.system() == "Windows":
-            self._denoiser_path += ".exe"
+        self._denoiser_path = which(
+            "denoise",
```
Comment (Theverat) [inline draw/viewport.py:78]: Shouldn't this be `oidnDenoise` as well?

**Comment by Theverat** [comment]
> For release standpoint, it makes no sense at all, but as the packager, I have multiple packages depending on openimagedenoise and have to patch LuxBlendCore to be able to work with system-wide openimagedenoise.
If it would be more acceptable, I could switch around ordering - prepending CWD/bin to make local denoise goes first

That sounds reasonable. So your packaging script only has to delete our copy of the OIDN binary.

File: draw/viewport.py (hunk 1 above)
Comment (bartoszek) [reply draw/viewport.py]: Fixed.

**Comment by bartoszek** [comment]
I've updated `openimagedenoise>=1.2.1` patch to include `bin/.gitignore` and `bin/get_binary.py`

**Comment by Theverat** [comment]
Thanks!

**Pull request merged**